File: atcoder/abc185A_abc_preparation.c
// uninhm
// https://atcoder.jp/contests/abc185/tasks/abc185_a
// implementation

#include <stdio.h>

int main() {
  int a[4];
  scanf("%d %d %d %d", &a[0], &a[1], &a[2], &a[3]);

  int ans = 1 << 30;
  for(int i = 0; i < 4; ++i)
    ans = a[i] < ans ? a[i] : ans;

  printf("%d\n", ans);
}

File: atcoder/abc185A_abc_preparation.rs
// vicfred
// https://atcoder.jp/contests/abc185/tasks/abc185_a
// implementation
use std::io;
use std::cmp::min;

fn main() {
    let mut line = String::new();

    io::stdin()
        .read_line(&mut line)
        .unwrap();

    let a: Vec<i64> =
        line
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let ans = min(min(a[0], a[1]), min(a[2], a[3]));

    println!("{}", ans);
}

